packages/connect-web/src/impl/core-in-iframe.ts

```diff
@@ -174,6 +173,18 @@ export class CoreInIframe implements ConnectFactoryDependencies {
 
         await iframe.init(this._settings);
 
+        if (this._settings.coreMode === 'auto') {
+            // Checking bridge availability
+            console.log('Checking bridge availability'); // DEBUG
+            const { promiseId, promise } = this._messagePromises.create();
+            iframe.postMessage({ id: promiseId, type: TRANSPORT.GET_INFO });
+            const response = await promise;
+            console.log('Bridge availability response', response);
+            if (response.payload === undefined) {
+                throw ERRORS.TypedError('Transport_Missing');
+            }
+        }
+
```

Comment:
I am still thinking about the race-condition scenario. not sure if we were able to rule it out or not. the worst thing that can happen is that under mode `auto` we will switch to popup mode even when it is not necessary, right?

Reply:
Yes, nothing too bad should happen

Reply:
But I actually changed `transportReconnect` to wait in case of `auto` mode